### PR TITLE
[Tooling] [L10n] Add Jetpack-specific strings to localization pipeline

### PR DESCRIPTION
**Pull request by AliSoftware**
This PR adds the `WordPress/src/jetpack/res/values*/strings.xml` to the translation pipeline lanes

## Why?

Ref: pdnsEh-pY-p2 

So far the WordPress and Jetpack app both only used `Wordpress/src/main/res/values*/strings.xml` as their shared set of common strings, since both apps are built on the same codebase and shared the same strings. So we only sent those strings.xml files to GlotPress for translation and back.

Since recently though, we've started to need Jetpack-specific strings for the Jetpack app to override some strings from the WordPress variant, and used `WordPress/src/jetpack/res/values*/strings.xml` for that. But those new files had never been added to our translation pipeline, which means they never been sent to GlotPress to get translated and back (and indeed we only have `WordPress/src/jetpack/res/values/` but no other locale-specific `values-{locale}` folder there so far). This PR fixes that.

## What's Next

In addition to this, I have submitted a diff to update our cron job importing the new frozen jetpack file into GlotPress. See D83569-code

## To Test

Start by cutting a test branch from this PR's branch, to run the following tests

### Checking Translations Coverage

* Run `bundle exec fastlane check_translations_coverage`
* Reply `y` each 4 times it asks you if you want to continue.
* Confirm that the output shows that all locales for the Jetpack app strings are 0%

### Freezing Strings

* Edit `WordPress/src/jetpack/res/values/strings.xml` to change a value of one of the strings (or add a new entry)
* Run `bundle exec fastlane update_frozen_strings_for_translation`
* Verify that the lane created a commit containing a change to the `fastlane/jetpack_resources/values/strings.xml` file with the change you made to the main file

### Downloading App Translations

* _[Optional] Edit `fastlane/lanes/localization.rb` and replace the value for the two `lint_task:` parameter on line 395 and 402 with empty strings. That will allow you to avoid waiting forever for the lint to run when the goal is only to test the lane._
* Run `bundle exec fastlane download_translations`
* Ensure that it downloads the latest translations for WordPress… and tries to do so for Jetpack as well — but because it's currently at 0%, won't find anything to download and won't create the files for now.
* Especially, ensure that the logs show the right list of locales it tries to download for the Jetpack app strings (i.e. our Mag16), which is not the same subset as the locales we have for WordPress app strings (where we have way more)

### Downloading Metadata Translations

* Run `bundle exec fastlane download_metadata_strings version:20.2 build_number:1241`
* Validate it still downloads both WP and JP metadata translations like it always has. Note that there might not be many new metadata in GlotPress at the time, and thus (m)any `.txt` file modified in that created commit — except the old changelog being deleted

### CODE REVIEW

File: fastlane/lanes/localization.rb
Comment (AliSoftware) [inline fastlane/lanes/localization.rb:395]: TODO in a future PR: Remove the call to the lint tasks once we finally migrate the "Release Build" jobs from CircleCI to Buildkite and can make the "Release Build" jobs in Buildkite `depend_on` the Buildkite lint job.

_The migration of those jobs to Buildkite was expected to be done way sooner which is why those have still be there for a while, but hopefully I plan to work on the migration of those to Buildkite very soon — which will allow me to save a lot of time during releases by not having to wait for two lint tasks to finish running on my Mac before CI starts (which often meant that I couldn't do any other work on WPAndroid while waiting for it to finish running)_

File: fastlane/lanes/localization.rb
Comment (mokagio) [inline fastlane/lanes/localization.rb:339]: Have you considered doing two commits? Would it be useful to be able to look at the diff for only one app at a time?

**Review by mokagio — APPROVED** [review]
> **Checking Translations Coverage**
> Run `bundle exec fastlane check_translations_coverage`
> ...
Done. Saw all 4 checks.

> Confirm that the output shows that all locales for the Jetpack app strings are 0%

image

> **Freezing Strings**
> ...

✅ [wordpress-mobile/WordPress-Android@`fc4d6f6` (#16865)](https://github.com/wordpress-mobile/WordPress-Android/pull/16865/commits/fc4d6f62891816039ee2b5ef273a729f508c78b5)

> **Downloading App Translations**
> ...
> Ensure that it [`bundle exec fastlane download_translations`] downloads the latest translations for WordPress… and tries to do so for Jetpack as well...

✅

> ...— but because it's currently at 0%, won't find anything to download and won't create the files for now.

🤔 we actually got files created, but empty see [wordpress-mobile/WordPress-Android@`a150df5` (#16865)](https://github.com/wordpress-mobile/WordPress-Android/pull/16865/commits/a150df5386b7a67bfb1d9fa761b1166e5c2f2b7c)

> Especially, ensure that the logs show the right list of locales it tries to download for the Jetpack app strings (i.e. our Mag16), which is not the same subset as the locales we have for WordPress app strings (where we have way more)

✅ 
image

> **Downloading Metadata Translations**
> ...

✅ 

---

The `strings.xml` download didn't behave as described, but the behavior I experienced (creating valid XMLs but with no string nodes) seems acceptable to me, and it's anyways only a temporary state till there will be translations.

This PR doesn't have auto-merge, so I'm approving to unblock and you can decide whether it's okay to merge–I obviously think it is, but I might be missing something.

File: fastlane/lanes/localization.rb
Comment (AliSoftware) [reply fastlane/lanes/localization.rb:339]: I thought about it (in fact that's what I did with my initial implementation, splitting this lane in two sublanes, and having the current lane calling both)… but considered it wasn't really worth it as (1) it created more noise both in the Fastfile code and in the commit list (2) since the files modified by each case are separate, having 2 commits each containing one file would not benefit us much as opposed to one commit with two distinct files, because the changes on each of the two files are already independant, so if we want to look at the diff for only one app at a time we already can, by just looking at the corresponding `.xml` file.

**Comment by AliSoftware** [comment]
> This PR doesn't have auto-merge, so I'm approving to unblock and you can decide whether it's okay to merge–I obviously think it is, but I might be missing something.

Mostly because I wanted to be sure to manually coordinate the landing of this PR with the corresponding Phabricator diff D83569-code. Now that both are approved, this is ready to land.

@mokagio I've thought of a limitation of all of this though, which is that keys meant to be overridden but that would be missing a translation in one of the `jetpack/values-{locale}/strings.xml` locale… would still fallback to the translation from `main/values-{locale}/strings.xml` for that same locale. See pxLjZ-7b9-p2 for the whole discussion about this specific issue.

But:

1. This was already the case before this PR (since before the PR, we were missing translation for ALL the `jetpack/values-{locale}/strings.xml` keys, but now we will only miss the ones that our translation vendor would not have translated _yet_). So this PR is still an improvement over the _status quo_.
2. Hopefully for our Mag16, which are translated by our vendor, we'll have 100% coverage for those Jetpack-specific strings (there are not _that_ many so far anyway), so that shouldn't really be an issue in practice, at least at the beginning (until we have a potentially larger divergence between WP and JP strings)
3. So the main issue would be for locales that are translated in WP but not in JP, i.e. the non-Mag16 locales. But again, that's already a problem before this PR, so I'm planning to address this in a separate PR, probably by providing symlinks at least as a temporary workaround for those locales in Jetpack, to not block this PR and the Phabricator diff any longer.

File: fastlane/lanes/localization.rb
Comment (mokagio) [reply fastlane/lanes/localization.rb:339]: 👍

**Pull request merged**
